agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -166,8 +165,13 @@ def ensure_desired_state
       else
         warn "unknown state #{service_pod.desired_state} for #{service_pod.name}"
       end
+
+      service_container = get_container(service_pod.service_id, service_pod.instance_number)
+      service_container.name if service_container # trigger cached_json
+      service_container
     end
 
+    # @return [Docker::Container]
```

Comment:
Not using the `Docker::Container` that gets returned by the `Kontena::ServicePods::Creator`, because the way that the `cached_json` vs `state` stuff works is too confusing. The `container.info` is from what `Docker::Container.create` returns, which would be from before the call to `start!`. What the various container methods return then depends on if something triggers `cached_json` before or after the call to `start!` - don't want to deal with that, just inspect the container again.

Doesn't really matter for `service_container.running?` though, because that re-queries the Docker API (twice) for each call.